blog/2023-05-18-how-to-use-conda-build.mdx

```diff
@@ -0,0 +1,122 @@
+
+
+
+### Installing conda-build
+ Create a new [conda environment](https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/environments.html) and run the following command on your terminal :
+```bash
+conda install conda-build
+```
+
+### Creating a template for the recipe
+For this tutorial, we'll be walking you through the process of building a recipe for PyPi's "click" package. Pypi is among the various repositories available for python softwares on the internet. 
+```bash
+conda skeleton pypi click
+```
+:::note
+
+Use can use any package available from PyPi instead of click
+
+:::
+
+
+After executing this command, a new directory called "click" will be generated. Inside this directory, you'll find a file named "meta.yaml". This file contains all the metadata necessary to create a conda package, such as the package name, version number, source code address, installation instructions, dependencies, hosting location, license, and more. Note that if you didn't use the "skeleton" command, you'll need to manually write the script yourself. You can make  any changes to meta.yaml file and be the maintainer and distributor of your package.
+
+
+## Building the package
+Run this command from the root folder :
+```bash
+conda-build click
+```
+
+After the build process is finished,  conda-build will generate a package file in the 'conda-bld' directory.
+To locate this  file, you can use this command from the terminal:
+```bash 
+conda build click --output
+```
+You can now install the package locally by running this command in the terminal.
+```bash 
+ conda install --use-local click
+ ```
+ Congrats 🎊🎉, you have successfully downloaded your own package! You can modify this package and upload it to Anaconda cloud.
+
+### Building a package with C/C++ dependency
+With conda-build, it is extremely simple to include non-python packages.
+
+In the "meta.yaml" file you can include the following and build the package again:
+```bash
+script: 
+    g++ -o mylib.so -shared -fPIC mylib.cpp
+
+requirements:
+  build:
+    - gcc  
+ run:
+    - mylib 
+ ```   
+
+In this example, we have added gcc as a build dependency, and mylib as a runtime dependency. We have also added a new step to compile mylib.cpp into a shared library called mylib.so.
+
+## Testing the package
+After successfully creating our package, we can now proceed to create another conda environment on our machine and use the newly created package. Additionally, we can also upload the package to Anaconda Cloud and distribute it to others or create our channel.
+
+To install the package on your local machine, use the following command:
+
+```bash
+conda create -n test_env
+conda activate test_env
+conda install click
+```
+
+You can also upload your package to any existing channel and distribute it to others. You can also create your own channel.
```

Comment:
Maybe go with something like: "Use the following command to upload your package to Anaconda.org:"

And then after the bash command below: "You can upload your package to any existing channel, including one you create for yourself, and distribute it to others."

Maybe a link to the anaconda.org documentation for more details?

Reply:
Sure, I will do that and add the link too.